File: streaming-server/src/main/java/io/github/hs96wings/streaming_server/chat/domain/ChatParticipant.java
package io.github.hs96wings.streaming_server.chat.domain;

import io.github.hs96wings.streaming_server.common.domain.BaseTimeEntity;
import io.github.hs96wings.streaming_server.member.domain.Member;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Entity
@NoArgsConstructor
@AllArgsConstructor
@Builder
@Getter
public class ChatParticipant extends BaseTimeEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "chat_room_id", nullable = false)
    private ChatRoom chatRoom;
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "member_id", nullable = false)
    private Member member;
}

File: streaming-server/src/main/java/io/github/hs96wings/streaming_server/chat/domain/ChatParticipant.kt
package io.github.hs96wings.streaming_server.chat.domain

import io.github.hs96wings.streaming_server.common.domain.BaseEntity
import io.github.hs96wings.streaming_server.member.domain.Member
import jakarta.persistence.*

@Entity
open class ChatParticipant @JvmOverloads constructor(
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "chat_room_id", nullable = false)
    var chatRoom: ChatRoom,

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "member_id", nullable = false)
    var member: Member
): BaseEntity() {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    override val id: Long? = null

    @OneToMany(mappedBy = "chatParticipant", cascade = [CascadeType.REMOVE], orphanRemoval = true)
    var readStatuses: MutableList<ReadStatus> = mutableListOf()
}